### ISSUE TITLE
Using a common load balancer between two EKS Clusters(Fargate and Non-Fargate)

### ISSUE THREAD

**Issue by nitinkapur** [kubernetes-sigs/aws-load-balancer-controller#1613]
I am trying to set up a common load balancer to serve routes between different EKS clusters one cluster on Fargate and another traditional EKS cluster with EC2 nodes.

The use case is to have one DNS name shared between different applications distributed on different EKS clusters and routing based on paths.

I first manually created an internal application load balancer and a target group using the aws console. In load balancer all the routes "/*" are being routed to the target group. If I make my service of the type nodeport and use targetbindinggroup to make that service part of the target group it doesn't works. I have noticed that even the nodeport service doesn't seems to work on it's own on fargate, maybe because the hosts work differently in fargate. However, I do see a new target being registered on the target group when I kubectl apply the targetgroupbinding file.

If I try to get nodes using
```
kubectl get nodes
```
and use one of the nodes with the nodeport service's exposed port number like
```
http://10.12.132.66:31015
```
and hit the service. I do not get any reply and request times out.

Now, I am trying to set it up as an ip. This is how I am defining the service
```
apiVersion: v1
kind: Service
metadata:
  name: my-service
  labels:
    app: my-service
   annotations:
     alb.ingress.kubernetes.io/target-type: ip
spec:
  selector:
    app: my-service
  ports:
  - name: http-my-service
    protocol: "TCP"
    port: 80
    targetPort: 80
```
this is how I am defining the targetgroupbinding

```
apiVersion: elbv2.k8s.aws/v1alpha1
kind: TargetGroupBinding
metadata:
  name: someName
  namespace: dev
spec:
  networking:
    ingress:
    - from:
      - securityGroup:
          groupID: _hidden_
      ports:
      - protocol: TCP
  serviceRef:
    name: my-service
    port: 80
  targetGroupARN: _hidden_
  targetType: ip
```

If I do so I get the below error from AWS load balance controller
```
"ValidationError: Instance ID '10.133.106.251' is not valid\n\tstatus code: 400
```

Why is it still trying to set up Instance ID when I am specifying the targettype as ip?

**Comment by M00nF1sh** [comment]
@nitinkapur 
The TargetType is defined in AWS when you create the TargetGroup in AWS. (you need to create a IP target type TargetGroup in AWS if you want to use the pod ip targeting).

The TargetType in TargetGroupBinding don't need to be set, it will be set correctly by webhook according to TargetGroup's TargetType in AWS. (if you set it manually, it need to match the actual target type in aws)

**Comment by nitinkapur** [comment]
Thanks! I should have mentioned that while creating target group I selected instance type. You were still able to figure it out, that's smart!

So, I fixed it and created a target group which is "ip" type with the tags. 
![image](https://user-images.githubusercontent.com/10778343/98238087-e0307f80-1f33-11eb-9eb4-31348480dc40.png)

Then I set up the below forwarding rule in the load balancer

![image](https://user-images.githubusercontent.com/10778343/98237810-7dd77f00-1f33-11eb-8dd9-5a98ceefbcbf.png)

after that I deployed the targetgroupbinding without targetType and got the below error.

```
{"level":"error","ts":1604576595.158312,"logger":"controller","msg":"Reconciler error","reconcilerGroup":"elbv2.k8s.aws","reconcilerKind":"TargetGroupBinding","controller":"targetGroupBinding","name":"worklink-otss-tgrpb","namespace":"test","error":"admission webhook \"vtargetgroupbinding.elbv2.k8s.aws\" denied the request: TargetGroupBinding must specify these fields: spec.targetType"}
```

So I guess targetType is required which I added.
```
apiVersion: elbv2.k8s.aws/v1alpha1
kind: TargetGroupBinding
metadata:
  name: worklink-otss-tgrpb
  namespace: test
spec:
  networking:
    ingress:
    - from:
      - securityGroup:
          groupID: sg-057b06b7333624e01
      ports:
      - protocol: TCP
  serviceRef:
    name: ganesh-permissions
    port: 80
  targetGroupARN: arn:aws:elasticloadbalancing:us-east-1:152032075715:targetgroup/otss-target-group-test/fbe9b5f1d034373b
  targetType: ip
```
then my target got registered. 
```
{"level":"info","ts":1604576779.489685,"msg":"registering targets","arn":"arn:aws:elasticloadbalancing:us-east-1:152032075715:targetgroup/otss-target-group-test/fbe9b5f1d034373b","targets":[{"AvailabilityZone":null,"Id":"10.133.88.55","Port":80}]}
{"level":"info","ts":1604576780.004092,"msg":"registered targets","arn":"arn:aws:elasticloadbalancing:us-east-1:152032075715:targetgroup/otss-target-group-test/fbe9b5f1d034373b"}
```
and I can see that it in the aws console as well.

![image](https://user-images.githubusercontent.com/10778343/98238429-5c2ac780-1f34-11eb-85d2-4042705836b1.png)

so far so good. But I am still getting a request timeout error. The service is responding fine if I curl from within the cluster. Which is returning back it's version at the root

```
kubectl exec -n test -it ganesh-permissions-test-78b57cff8f-z8hjx  curl http://ganesh-permissions
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl kubectl exec [POD] -- [COMMAND] instead.
{"version":"2.7.1"}%
```
Is there anything else that I am missing? On securitygroup of targetgroupbinding in inbound rules all connections from 0.0.0.0 are allowed. What could be the reason for this timeout then?

**Comment by nitinkapur** [comment]
So I figured out the "Request time out error" it was due to the security group not set properly.

I have now created a nodeport service in another cluster and registered the targets as all the EC2 instances in that cluster.

![image](https://user-images.githubusercontent.com/10778343/98293955-2231e380-1f7d-11eb-9701-62ca969f18f9.png)

I am now getting a 404 error. This is how my path's are set. All my services are serving from path "/"

This is ow they are setup

![image](https://user-images.githubusercontent.com/10778343/98294153-750b9b00-1f7d-11eb-9762-c6d48fa08df4.png)

service in the target group "k8s-test-ingressg-21a43e5770" is returning the version fine but the service in "otss-tgrp" is throwing up 404.

![image](https://user-images.githubusercontent.com/10778343/98294299-b9973680-1f7d-11eb-8046-d301774345db.png)

![image](https://user-images.githubusercontent.com/10778343/98294332-cae04300-1f7d-11eb-8e22-027be2186fc5.png)
